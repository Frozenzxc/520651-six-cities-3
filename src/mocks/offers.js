import {getRandomArrayItem, getRandomIntegerNumber} from "../common";

const MAX_BEDROOMS = 6;
const MIN_BEDROOMS = 1;
const MAX_GUESTS = 10;
const MAX_PHOTOS = 6;
const MAX_RATING = 5;
const MIN_RATING = 1;

const CardTypes = [`apartment`, `room`, `house`, `hotel`];
const Options = [`Wifi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`];
const Photos = [
  `img/apartment-01.jpg`,
  `img/apartment-02.jpg`,
  `img/apartment-03.jpg`,
  `img/apartment-01.jpg`,
  `img/apartment-02.jpg`,
  `img/apartment-03.jpg`,
];

const generatePhotos = (photos) => {
  return photos
        .filter(() => Math.random() > 0.5)
        .slice(0, getRandomIntegerNumber(1, MAX_PHOTOS));
};

const generateOptions = (options) => {
  return options
        .filter(() => Math.random() > 0.5)
        .slice(0, getRandomIntegerNumber(0, Options.length));
};

export default [
  {
    bedroomsCount: 4,
    coordinates: [52.3909553943508, 4.85309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      name: `John`,
      avatar: `img/avatar-max.jpg`,
      super: false,
    },
    id: 1,
    maxGuests: 6,
    options: [`Wifi`, `Washing machine`, `Towels`, `Heating`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    premium: false,
    price: 200,
    rating: 3,
    src: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious apartment at great location`,
    type: `house`,
  },
  {
    bedroomsCount: 2,
    coordinates: [52.369553943508, 4.85309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      name: `Max`,
      avatar: `img/avatar-angelina.jpg`,
      super: false,
    },
    id: 2,
    maxGuests: 4,
    options: [`Wifi`, `Towels`, `Heating`, `Cabel TV`, `Fridge`],
    premium: true,
    price: 300,
    rating: 1,
    src: [`img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Wood and stone place`,
    type: `hotel`,
  },
  {
    bedroomsCount: 1,
    coordinates: [52.3909553943508, 4.929309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      name: `Bob`,
      avatar: `img/avatar-max.jpg`,
      super: true,
    },
    id: 3,
    maxGuests: 2,
    options: [`Wifi`, `Towels`, `Cabel TV`],
    premium: false,
    price: 450,
    rating: 3,
    src: [`img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Canal View Prinsengracht`,
    type: `room`,
  },
  {
    bedroomsCount: 2,
    coordinates: [52.3809553943508, 4.939309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      super: false,
    },
    id: 4,
    maxGuests: 5,
    options: [`Wifi`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Cabel TV`],
    premium: false,
    price: 150,
    rating: 5,
    src: [`img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Nice, cozy, warm big bed apartment`,
    type: `apartment`,
  },
];
