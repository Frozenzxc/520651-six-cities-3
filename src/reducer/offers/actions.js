const ActionType = {
  BLOCK_FORM: `BLOCK_FORM`,
  LOAD_NEARBY_OFFERS: `LOAD_NEARBY_OFFERS`,
  LOAD_OFFERS: `LOAD_OFFERS`,
  LOAD_REVIEWS: `LOAD_REVIEWS`,
  SELECT_CARD: `SELECT_CARD`,
  SELECT_CITY: `SELECT_CITY`,
  SELECT_OFFER: `SELECT_OFFER`,
  SUCCESSFUL_POST_REVIEW: `SUCCESSFUL_POST_REVIEW`,
};

const ActionCreator = {

  blockForm: (isBlocked) => {
    return {
      type: ActionType.BLOCK_FORM,
      payload: isBlocked,
    };
  },

  loadNearbyOffers: (offers) => {
    return {
      type: ActionType.LOAD_NEARBY_OFFERS,
      payload: offers,
    };
  },

  loadOffers: (offers) => {
    return {
      type: ActionType.LOAD_OFFERS,
      payload: offers,
    };
  },

  loadReviews: (reviews) => {
    return {
      type: ActionType.LOAD_REVIEWS,
      payload: reviews,
    };
  },

  selectCard: (offer) => ({
    type: ActionType.SELECT_CARD,
    payload: offer.id,
  }),

  selectCity: (city) => ({
    type: ActionType.SELECT_CITY,
    payload: city,
  }),

  selectOffer: (offer) => ({
    type: ActionType.SELECT_OFFER,
    payload: offer,
  }),

  successfulPostReview: (isSuccess) => ({
    type: ActionType.SUCCESSFUL_POST_REVIEW,
    payload: isSuccess,
  })

};

export {ActionType, ActionCreator};
