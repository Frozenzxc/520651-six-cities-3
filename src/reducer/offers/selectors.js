import {createSelector} from "reselect";
import NameSpace from "../name-space";
import {MAX_NEARBY_OFFERS} from "../../const";

const getCurrentCity = (state) => {
  return state[NameSpace.OFFERS].currentCity;
};

const getActiveOffer = (state) => {
  return state[NameSpace.OFFERS].activeOffer;
};

const getFavoriteOffers = (state) => {
  return state[NameSpace.OFFERS].favoriteOffers;
};

const getNearbyOffers = (state) => {
  return state[NameSpace.OFFERS].nearbyOffers.slice(0, MAX_NEARBY_OFFERS);
};

const getOffers = (state) => {
  return state[NameSpace.OFFERS].offers;
};

const getReviews = (state) => {
  return state[NameSpace.OFFERS].reviews;
};

const getAvailableOffers = createSelector(
    getOffers,
    getCurrentCity,
    (offers, currentCity) => {
      return offers.filter((offer) => offer.city.name === currentCity);
    }
);

export {getCurrentCity, getActiveOffer, getNearbyOffers, getOffers, getReviews, getAvailableOffers, getFavoriteOffers};
