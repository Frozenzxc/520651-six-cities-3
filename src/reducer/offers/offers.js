import {extend} from "../../utils.js";
import {parseOffer, parseReview} from "../../utils";
import {ActionCreator, ActionType} from "../offers/actions";
import {ReviewPostingStatus} from "../../const";

const getAvailableOffers = ((allOffers, currentCity) => allOffers.filter((offer) => offer.city.name === currentCity));

const initialState = {
  offers: [],
  availableOffers: [],
  activeID: null,
  activeOffer: null,
  currentCity: `Amsterdam`,
  isFormBlocked: false,
  isLoading: true,
  isPropertyLoading: true,
  nearbyOffers: [],
  reviews: [],
  reviewPostingStatus: null,
};

const Operation = {
  loadNearbyOffers: (id) => (dispatch, getState, api) => {
    return api.get(`/hotels/${id}/nearby`)
            .then((response) => {
              dispatch(ActionCreator.loadNearbyOffers(response.data));
            });
  },

  loadOffers: () => (dispatch, getState, api) => {
    return api.get(`/hotels`)
            .then((response) => {
              dispatch(ActionCreator.loadOffers(response.data));
            });
  },

  loadReviews: (id) => (dispatch, getState, api) => {
    return api.get(`/comments/${id}`)
            .then((response) => {
              dispatch(ActionCreator.loadReviews(response.data));
            });
  },

  postReview: (id, formData) => (dispatch, getState, api) => {
    return api.post(`/comments/${id}`, {
      comment: formData.comment,
      rating: formData.rating,
    })
          .then((response) => {
            dispatch(ActionCreator.successfulPostReview(ReviewPostingStatus.POSTED));
            dispatch(ActionCreator.loadReviews(response.data));
            dispatch(ActionCreator.blockForm(false));
          })
        .catch((err) => {
          dispatch(ActionCreator.successfulPostReview(ReviewPostingStatus.ERROR));
          dispatch(ActionCreator.blockForm(false));
          throw err;
        });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.BLOCK_FORM:
      return extend(state, {
        isFormBlocked: action.payload,
      });

    case ActionType.LOAD_NEARBY_OFFERS:
      let parsedNearbyOffers = action.payload.map((offer) => parseOffer(offer));
      return extend(state, {
        nearbyOffers: parsedNearbyOffers,
      });

    case ActionType.LOAD_OFFERS:
      let parsedOffers = action.payload.map((offer) => parseOffer(offer));
      return extend(state, {
        currentCity: parsedOffers[0].city.name,
        isLoading: false,
        offers: parsedOffers,
      });

    case ActionType.LOAD_REVIEWS:
      let parsedReviews = action.payload.map((review) => parseReview(review));
      return extend(state, {
        isPropertyLoading: false,
        reviews: parsedReviews,
      });

    case ActionType.SELECT_CARD:
      return extend(state, {
        activeID: action.payload,
      });

    case ActionType.SELECT_CITY:
      return extend(state, {
        currentCity: action.payload,
        availableOffers: getAvailableOffers(state.offers, action.payload),
      });

    case ActionType.SELECT_OFFER:
      return extend(state, {
        activeOffer: action.payload,
      });

    case ActionType.SUCCESSFUL_POST_REVIEW:
      return extend(state, {
        reviewPostingStatus: action.payload,
      });
  }

  return state;
};

export {reducer, Operation, ActionType, ActionCreator};
