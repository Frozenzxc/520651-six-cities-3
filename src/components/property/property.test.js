import React from 'react';
import renderer from 'react-test-renderer';
import Property from "./property";
import offers from "../../test-mocks/test-offers";

const offer = offers[0];

it(`Property is rendered correctly`, () => {
  const tree = renderer.create(
      <Property
        offer={offer}
        offers={offers}
        onCardHover={() => {}}
        onCardTitleClick={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
