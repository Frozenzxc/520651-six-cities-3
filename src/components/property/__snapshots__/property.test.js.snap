// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property is rendered correctly 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            The house among olive 
          </h1>
          <button
            className="property__bookmark-button button"
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "96%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            4.8
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            apartment
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            3
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            7
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            160
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Coffee machine
            </li>
            <li
              className="property__inside-item"
            >
              Laptop friendly workspace
            </li>
            <li
              className="property__inside-item"
            >
              Fridge
            </li>
            <li
              className="property__inside-item"
            >
              Breakfast
            </li>
            <li
              className="property__inside-item"
            >
              Air conditioning
            </li>
            <li
              className="property__inside-item"
            >
              Towels
            </li>
            <li
              className="property__inside-item"
            >
              Washer
            </li>
            <li
              className="property__inside-item"
            >
              Dishwasher
            </li>
            <li
              className="property__inside-item"
            >
              Baby seat
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.
            </p>
            <p
              className="property__text"
            >
              Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.
            </p>
          </div>
        </div>
      </div>
    </div>
    <section
      className="property__map map"
    >
      <div
        id="map"
        style={
          Object {
            "height": "100%",
          }
        }
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list tabs__content"
      >
        <article
          className="near-places__card place-card"
          onMouseEnter={[Function]}
        >
          <div
            className="near-image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  192
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "84%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
              onClick={[Function]}
            >
              <a
                href="#"
              >
                Waterfront with extraordinary view
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              room
            </p>
          </div>
        </article>
        <article
          className="near-places__card place-card"
          onMouseEnter={[Function]}
        >
          <div
            className="near-image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  350
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "88%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
              onClick={[Function]}
            >
              <a
                href="#"
              >
                Nice, cozy, warm big bed apartment
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              hotel
            </p>
          </div>
        </article>
        <article
          className="near-places__card place-card"
          onMouseEnter={[Function]}
        >
          <div
            className="near-image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  196
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "66%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
              onClick={[Function]}
            >
              <a
                href="#"
              >
                Loft Studio in the Central Area
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              house
            </p>
          </div>
        </article>
        <article
          className="near-places__card place-card"
          onMouseEnter={[Function]}
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="near-image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  116
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "84%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
              onClick={[Function]}
            >
              <a
                href="#"
              >
                Penthouse, 4-5 rooms + 5 balconies
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              apartment
            </p>
          </div>
        </article>
        <article
          className="near-places__card place-card"
          onMouseEnter={[Function]}
        >
          <div
            className="near-image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  892
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "60%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
              onClick={[Function]}
            >
              <a
                href="#"
              >
                Loft Studio in the Central Area
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              house
            </p>
          </div>
        </article>
        <article
          className="near-places__card place-card"
          onMouseEnter={[Function]}
        >
          <div
            className="near-image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  136
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
              onClick={[Function]}
            >
              <a
                href="#"
              >
                Waterfront with extraordinary view
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              apartment
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;
